evidence/pool.go

```diff
@@ -523,19 +399,26 @@ func (evpool *Pool) listEvidence(prefixKey byte, maxBytes int64) ([]types.Eviden
 		return nil, totalSize, fmt.Errorf("database error: %v", err)
 	}
 	defer iter.Close()
+	var protoOverheadBytes int64 = 3
 	for ; iter.Valid(); iter.Next() {
-		evInfo, err := bytesToInfo(iter.Value())
+		var evpb tmproto.Evidence
+		err := evpb.Unmarshal(iter.Value())
 		if err != nil {
-			return nil, totalSize, err
+			return evidence, totalSize, err
+		}
+		evSize := int64(evpb.Size())
```

Comment:
doesn't this already include the proto overhead?

Reply:
Since you have `[]evidence`, a simpler approach could be to add the new evidence to `Evidencelist` and call size on it, this way you won't have to hardcode a `protoOverheadBytes` variable.

Reply:
But I would have to calculate the entire size over and over again instead of just incrementing the size of the next evidence

Reply:
yea, it's a tradeoff, but gets you the exact size each time. The overhead for calculating it each time should be quite small. 

I think the calculation below is incorrect, you are adding extra bytes on every evidence, but it would only be needed on the EvidenceList message.

Reply:
yeah I reversed engineered this from a test. I think the three bytes comes actually from using `sum of` i.e. when we convert from the concrete types to the evidence interface.

Reply:
I guess this could change then when we add more evidence to the proto evidence "interface"